docs/index.md

```diff
@@ -199,6 +199,7 @@ Type: [Object][29]
 -   `address` **[Buffer][42]** the address of the `account`
 -   `depth` **[Number][32]** the current number of calls deep the contract is
 -   `memory` **[Buffer][42]** the memory of the VM as a `buffer`
+-   `memoryWordCount` **BN** current size of memory in words
```

Comment:
Sorry, just noticed this was a BN instance. I think we should make this a `Number` by changing `memoryWordCount: runState.memoryWordCount` to `memoryWordCount: runState.memoryWordCount.toNumber()`.

It is highly unlikely the word count would be bigger 53 bits, i.e. 576460752303423456 bytes of memory :)

Reply:
Ah, wouldn't have thought on this, but - yes - we should take this extra round since this belongs to the public API and should be done right. @pinkiebell can you prepare a short follow-up PR?

Reply:
@axic @holgerd77

If we change that to a `Number`, then it will not be possible to reflect changes 🤔 .
Like `memoryWordCount.iaddn(1)`

Reply:
Why would that be needed? This is a step method, it should be used for retrieving information, not modifying it.

Reply:
@axic My intent with this pull request was to make it possible to shrink/growing memory inside step events. If you are strongly against it, then another solution could be to remove `memoryWordCount` references and change affected functions to always derive the size in words from `memory` array itself.

Reply:
Why would want to shrink memory?

Reply:
@axic We use it for https://github.com/leapdao/solEVM-enforcer .
We want to make use of ethereumjs-vm in our tooling and off-chain tools, so we need ways manipulate/grow/shrink memory or other runtime objects.
With ethereumjs-vm we also have a reference implementation to check against the solidity implementation.

Reply:
Ah, that's definitely not the way to go to use the ``step`` event for state manipulation, that adds a large attack surface on the API. We have to find another way here.

Reply:
Sounds good. For my use case it would be enough if I could pass initial stack values, initial memory and the ability to pass a initial `programCounter`. 

I'm also interested in tracking memory access inside step events. <- That's probably a bigger change 😄 

@holgerd77 I'm going to open a new issue for that soon and in the meantime you can revert my commits if you want.

Reply:
link #410